examples/sha512/src/sha512.rs

```diff
@@ -0,0 +1,162 @@
+// Import hacspec and all needed definitions.
+use hacspec_lib::*;
+
+const BLOCK_SIZE: usize = 128;
+const LEN_SIZE: usize = 16;
+pub const K_SIZE: usize = 80;
+pub const HASH_SIZE: usize = 512 / 8;
+
+bytes!(Block, BLOCK_SIZE);
+array!(OpTableType, 12, usize);
+bytes!(Sha512Digest, HASH_SIZE);
```

Comment:
We should indeed find a way to parameterize specs indeed to avoid having these copies.

Reply:
We probably want to go with generics. But let's discuss next week.
Something like we have in blake2 here would probably be nice https://github.com/hacspec/hacspec/blob/master/examples-unsafe/src/blake2/blake2b.rs